pallets/registrar/src/benchmarks.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+
+#![cfg(feature = "runtime-benchmarks")]
+
+//! Benchmarking
+use {
+    crate::{Call, Config, DepositBalanceOf, Pallet},
+    frame_benchmarking::{account, benchmarks},
+    frame_support::traits::Currency,
+    frame_system::RawOrigin,
+    sp_core::Get,
+    sp_std::vec,
+    tp_container_chain_genesis_data::ContainerChainGenesisData,
+    tp_traits::ParaId,
+};
+
+/// Create a funded user.
+/// Used for generating the necessary amount for registering
+fn create_funded_user<T: Config>(
+    string: &'static str,
+    n: u32,
+    extra: DepositBalanceOf<T>,
+) -> (T::AccountId, DepositBalanceOf<T>) {
+    const SEED: u32 = 0;
+    let user = account(string, n, SEED);
+    let min_reserve_amount = T::DepositAmount::get();
+    let total = min_reserve_amount + extra;
+    T::Currency::make_free_balance_be(&user, total);
+    T::Currency::issue(total);
+    (user, total)
+}
+
+benchmarks! {
+    where_clause { where T::SessionIndex: From<u32> }
+
+    register {
+        // We make it dependent on the size of the runtime
+        let x in 5..3_000_000;
```

Comment:
I assume this is poking at proof size, but the resulting benchmarks don't appear to show any proof-size-per-x. Am I missing something? Maybe `verify` should show that the expected runtime blob exists in storage with the expected size.